operator/v1/types_ingress.go

```diff
@@ -545,6 +545,20 @@ type AWSLoadBalancerParameters struct {
 	//
 	// +optional
 	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+
+	// subnets specifies the list of subnets for the load balancer to
+	// route traffic to. The values can be either a subnet ID or name.
+	// Each subnet must be from a different availability zones otherwise
+	// the load balancer will not be provisioned. Additionally, if a subnet
+	// is not found, the load balancer will not be provisioned. Updating
```

Comment:
Can I go from no value to some value without deleting or is deleting always necessary?

Reply:
While using this API - No. You must always delete to recreate in order to effectuate. It's mandatory, or nothing changes.

Under the hood, the `service.beta.kubernetes.io/aws-load-balancer-subnets` service annotation does support updating subnets for ELB only (not NLB), but because cluster admins can break service reconciles if they get it wrong, and it's hard to detect a failure event from the CCM, we chose to make deletion mandatory for a clear signal (which is `status.loadbalancer.ingress.hostname` getting set indicating whether a LB was succesfully provisioned). EP [PR](https://github.com/openshift/enhancements/pull/1595/files#diff-f8f6da2366b6e9f9eede942d6a878c8043690acb03727fad6bbf71d37c95fd92R218) talks about it too.